main.py

```diff
@@ -105,18 +117,22 @@ def main():
 
     device = torch.device("cuda" if use_cuda else "cpu")
```

Comment:
use_cuda is not defined. Add something like "use_cuda = not args.no_cuda and torch.cuda.is_available()"

Reply:
missing 'args.no_cuda' in parser